pkg/storage/image.go

```diff
@@ -553,9 +554,17 @@ func (svc *imageService) ResolveNames(imageName string) ([]string, error) {
 		}
 		return nil, err
 	}
+
 	domain, remainder := splitDockerDomain(imageName)
+	// this means the image is already fully qualified
 	if domain != "" {
-		// this means the image is already fully qualified
+		registry, err := sysregistriesv2.FindRegistry(nil, imageName)
+		if err != nil {
+			return nil, err
+		}
+		if registry != nil && registry.Blocked {
+			return nil, fmt.Errorf("cannot resolve names from registry %q because it's blocked", imageName)
```

Comment:
(Non-blocking nit: `imageName` is not a registry.  “cannot use %q because …”?)

Reply:
done